From c0c6e45ec7b338cd807dc0376d81f5fefb96aae8 Mon Sep 17 00:00:00 2001
From: Dan Pasanen <dan.pasanen@gmail.com>
Date: Sat, 11 Nov 2017 19:08:03 -0600
Subject: [PATCH 4/7] InputMethodManagerService: adjust grip mode for input
 enable/disable

* This is useful for mata (Essential PH-1) due to the super small bezel.

Change-Id: I9b8c9a8c6b866d03e1c3ff0975cb1c83f69fcaa0
---
 .../java/com/android/server/InputMethodManagerService.java     | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/services/core/java/com/android/server/InputMethodManagerService.java b/services/core/java/com/android/server/InputMethodManagerService.java
index c23757f..a06feb4 100644
--- a/services/core/java/com/android/server/InputMethodManagerService.java
+++ b/services/core/java/com/android/server/InputMethodManagerService.java
@@ -104,6 +104,7 @@ import android.os.RemoteException;
 import android.os.ResultReceiver;
 import android.os.ServiceManager;
 import android.os.SystemClock;
+import android.os.SystemProperties;
 import android.os.UserHandle;
 import android.os.UserManager;
 import android.provider.Settings;
@@ -249,6 +250,7 @@ public class InputMethodManagerService extends IInputMethodManager.Stub
     final HandlerCaller mCaller;
     final boolean mHasFeature;
     private InputMethodFileManager mFileManager;
+    private boolean mGripStatus;
     private final HardKeyboardListener mHardKeyboardListener;
     private final AppOpsManager mAppOpsManager;
     private final UserManager mUserManager;
@@ -2549,6 +2551,10 @@ public class InputMethodManagerService extends IInputMethodManager.Stub
             }
         }
 
+        if (mInputShown && mGripStatus) {
+            SystemProperties.set("sys.grip.status", "off");
+            mGripStatus = false;
+        }
         return res;
     }
 
@@ -2627,6 +2633,10 @@ public class InputMethodManagerService extends IInputMethodManager.Stub
         mShowRequested = false;
         mShowExplicitlyRequested = false;
         mShowForced = false;
+        if (res && !mGripStatus) {
+            SystemProperties.set("sys.grip.status", "on");
+            mGripStatus = true;
+        }
         return res;
     }
 
-- 
2.7.4

